Core/GDCore/Events/Parsers/GrammarTerminals.h

```diff
@@ -0,0 +1,107 @@
+
+inline bool IsQuote(gd::String::value_type character) {
+  return character == '"';
+}
+
+inline bool IsBracket(gd::String::value_type character) {
+  return character == '(' || character == ')' || character == '[' ||
+         character == ']' || character == '{' || character == '}';
+}
+
+inline bool IsOpeningParenthesis(gd::String::value_type character) {
+  return character == '(';
+}
+
+inline bool IsClosingParenthesis(gd::String::value_type character) {
+  return character == ')';
+}
+
+inline bool IsOpeningSquareBracket(gd::String::value_type character) {
+  return character == '[';
+}
+
+inline bool IsClosingSquareBracket(gd::String::value_type character) {
+  return character == ']';
+}
+
+inline bool IsExpressionEndingChar(gd::String::value_type character) {
+  return character == ',' || IsClosingParenthesis(character) ||
+         IsClosingSquareBracket(character);
+}
+
+inline bool IsExpressionOperator(gd::String::value_type character) {
+  return character == '+' || character == '-' || character == '<' ||
+         character == '>' || character == '?' || character == '^' ||
+         character == '=' || character == '\\' || character == ':' ||
+         character == '!';
+}
+
+inline bool IsUnaryOperator(gd::String::value_type character) {
+  return character == '+' || character == '-';
+}
+
+inline bool IsTermOperator(gd::String::value_type character) {
+  return character == '/' || character == '*';
+}
+
+inline bool IsNumberFirstChar(gd::String::value_type character) {
+  return character == '.' || (character >= '0' && character <= '9');
+}
+
+inline bool IsNonZeroDigit(gd::String::value_type character) {
+  return (character >= '1' && character <= '9');
+}
+
+inline bool IsZeroDigit(gd::String::value_type character) {
+  return character == '0';
+}
+
+/**
+ * Check if the given character can be used in an identifier. This is
+ * any unicode character, except for:
+ * `, . " () [] {} + - < > ? ^ = \ : ! / *` and whitespaces (space, line break, carriage return).
+ *
+ * This is loosely based on what is allowed in languages like JavaScript
+ * (see https://mathiasbynens.be/notes/javascript-properties), without support
+ * for unicode escape syntax, and allowing all unicode ranges. The only
+ * disallowed characters are the one used for the grammar.
+ */
+inline bool IsAllowedInIdentifier(gd::String::value_type character) {
+  // Quickly compare if the character is a number or ASCII character.
+  if ((character >= '0' && character <= '9') ||
+      (character >= 'A' && character <= 'Z') ||
+      (character >= 'a' && character <= 'z'))
+    return true;
+
+  // Otherwise do the full check against separators forbidden in identifiers.
+  if (!IsParameterSeparator(character) && !IsDot(character) &&
+      !IsQuote(character) && !IsBracket(character) &&
+      !IsExpressionOperator(character) && !IsTermOperator(character) &&
+      !IsWhitespace(character)) {
```

Comment:
Note that whitespace was added compared to the previous implementation. But this should not impact the parser.